src/data-access/repositories/groupRepository.ts

```diff
@@ -0,0 +1,79 @@
+import { DbError } from '../../core/errors/dbError.js';
+import { BaseGroup, Group } from '../../types/group.js';
+import { UserModel } from '../models/userModel.js';
+import { InitializeSequelize } from '../../database/postgreSQL/initializeSequelize.js';
+import { Model, Transaction } from 'sequelize';
+import { BaseRepository } from './baseRepository.js';
+
+
+export class GroupRepository extends BaseRepository {
+    constructor(groupModel: any, groupDataMapper: any) {
+        super(groupModel, groupDataMapper);
+        this.model = groupModel;
+        this.dataMapper = groupDataMapper;
+    }
+
+    async get(id: string): Promise<Group | undefined> {
+        const groupFromDB = await super.getEntityById(id);
+        return groupFromDB ? this.dataMapper.toDomain(groupFromDB.toJSON()) : undefined;
```

Comment:
repository should return RAW data. The transformation into the DTO format must be done on the service layer only . The reason is that, in some cases you need retrieved information for calculation, but in other cases it will be specifically for the client needs. So, that difference you will make on the service layer but not on the DB one(as repository is part of Database Layer).